crates/node/api/src/node.rs

```diff
@@ -26,32 +27,37 @@ pub trait NodeTypes: Send + Sync + Unpin + 'static {
     type Primitives: NodePrimitives;
     /// The node's engine types, defining the interaction with the consensus engine.
     type Engine: EngineTypes;
+    /// The type used for configuration of the EVM.
+    type ChainSpec: ChainSpecTrait;
```

Comment:
what's your argument for finding it appropriate to put `ChainSpec` here? I'd argue it's more appropriate as AT in `FullNodeTypes` trait, because looking at the other ATs contained in both traits `NodeTypes` and `FullNodeTypes`, `ChainSpec` semantically fits better into `FullNodeTypes` - the chain spec is more about initialisation of the chain, like db and provider, rather than compilation of the reth program for a specific network, like primitives and engine types.

Reply:
I don't have strong opinion here, so would be happy to move it. My reasoning was that `FullNodeTypes` keeps stateful components which are expected to change their behavior during node operation, while `NodeTypes` configures underlying types/configs for those components

Reply:
is the chain spec stateless? what about hardforks?

Reply:
> is the chain spec stateless? what about hardforks?

It can't change on its own during node operation (thus allowing us to keep it in an `Arc` everywhere). Primitive types can also change between hardforks